draft-ietf-jsonpath-base.md

```diff
@@ -1592,6 +1663,54 @@ Change controller:
 Provisional registration? (standards tree only):
 : no
 
+## Function Extensions {#iana-fnex}
+
+This specification defines a new "Function Extensions sub-registry" in
+a new "JSONPath Parameters registry", with the policy "expert review"
+({{Section 4.5 of -ianacons}}).
+
+The expert is instructed to be frugal in the allocation of very short
+function extension names, keeping them in reverse for applications that
+are likely to enjoy wide use and can make good use of their shortness.
+The expert is also instructed to direct the registrant to provide a
+specification ({{Section 4.6 of -ianacons}}), but can make exceptions,
+for instance when a specification is not available at the time of
+registration but is likely forthcoming.
+If the expert becomes aware of function extensions that are deployed and
+in use, they may also initiate a registration on their own if
+they deem such a registration can avert potential future collisions.
+{: #de-instructions}
+
+Each entry in the registry must include:
+
+{:vspace}
+Function Name:
+: a lower case ASCII {{-ascii}} string that starts with a letter and can
+  contain letters, digits and underscore characters afterwards
+  (`[a-z][_a-z0-9]*`).
+
+Brief description:
+: a brief description
+
+Input:
+: A comma-separated list of zero or more kinds (value or nodes) of the
+  arguments expected for this function extension
+
+Output:
+: The kind (value or nodes) of the result for this function extension
+
+Change Controller:
+: (see {{Section 2.3 of -ianacons}})
+
+Reference:
+: a reference document that provides a description of the function
+  extension
+
+Initial entries in this sub-registry are as listed in {{pre-reg}}:
+
+| Function Name | Brief description | Input | Output | Change Controller | Reference |   |   |
+| length        | length of array   | value | value  | IESG              | {{fnex}} of RFCthis |   |   |
```

Comment:
Do we need to allow an empty nodelist as input so that we can apply `length` to a `singular-path`?

Reply:
That should be a separate function (for a singular node list, it is not clear whether you want 1 or the length of the value in the node).

Reply:
ABNF says:

```
function-argument       = comparable
```

So literals are the other case.

Reply:
Separate function now in my proposal update.